client/fed_test.go

```diff
@@ -155,6 +155,42 @@ func TestGetAndPutAuth(t *testing.T) {
 		}
 	})
 
+	t.Run("testPelicanObjectPutAndGetWithQueryAndDestDir", func(t *testing.T) {
+		oldPref, err := config.SetPreferredPrefix(config.PelicanPrefix)
+		defer func() {
+			_, err := config.SetPreferredPrefix(oldPref)
+			require.NoError(t, err)
+		}()
+		assert.NoError(t, err)
+
+		// Set path for object to upload/download
+		for _, export := range fed.Exports {
+			tempPath := tempFile.Name()
+			fileName := filepath.Base(tempPath)
+			uploadURL := fmt.Sprintf("pelican://%s:%s%s/%s/%s", param.Server_Hostname.GetString(), strconv.Itoa(param.Server_WebPort.GetInt()),
+				export.FederationPrefix, "osdf_osdf", fileName)
+
+			// Upload the file with PUT
+			transferResultsUpload, err := client.DoPut(fed.Ctx, tempFile.Name(), uploadURL, false, client.WithTokenLocation(tempToken.Name()))
+			assert.NoError(t, err)
+			if err == nil {
```

Comment:
What's the logic for doing the `assert.NoError` and then checking `if err == nil`? Doesn't the first imply the second?

Reply:
Copy paste error, really. It's how it was in the original test, but you are correct.

Reply:
While we're in this section of code, can you remove the redundancy and do the same with any of the other tests that do it?